src/common/transformations/src/transformations/common_optimizations/ric_fusion.cpp

```diff
@@ -0,0 +1,607 @@
+                                                               create_const({1} /* output channel */));
+                input.replace_source_output(gather);
+                // TODO: copy runtime info from RIC sub-graph
+            });
+
+            if (auto fq = std::dynamic_pointer_cast<opset8::FakeQuantize>(conv->get_input_node_shared_ptr(1))) {
+                // Set final RIC attr to the first FQ input
+                ric_attr::set(fq->input(0), ric);
+
+                // Apply Binary transformation for FQ to handle 1..5 inputs
+                ric.set_is_final(false);
+                ric_attr::set(fq->input_value(0), ric); // set ric attr to simulate propagation flow
+                Binary().apply(fq);
+            } else {
+                ric_attr::set(conv->input(1), ric);
+            }
+            return true;
+        };
+
+        auto m = std::make_shared<pattern::Matcher>(pattern_root, matcher_name);
+        register_matcher(m, callback);
+    }
+};
+
+class GroupConvolution : public ngraph::pass::MatcherPass {
+public:
+    GroupConvolution() {
+        MATCHER_SCOPE(GroupConvolution);
+        auto input_p = pattern::any_input(ric_attr::has<Output<Node>>);
+        auto pattern_root = pattern::wrap_type<opset8::GroupConvolution>({input_p, pattern::any_input(pattern::has_static_shape())});
+
+        auto callback = [=](pattern::Matcher& m) {
+            auto conv = m.get_match_root();
+            const auto & weights_shape = conv->input_value(1).get_shape();
+            const int64_t & group = static_cast<int64_t>(weights_shape.at(0));
+            const int64_t & channels = static_cast<int64_t>(weights_shape.at(1));
+
+            auto ric = ric_attr::get(conv->input_value(0)).propagate();
+            if (ric.get_order().size() != static_cast<size_t>(group)) {
+                // TODO: insert RIC when group == 1
+                return false;
+            }
+
+            const int64_t output_channels = group * channels;
+            std::vector<int64_t> new_order;
+            new_order.reserve(output_channels);
+            for (const auto & index : ric.get_order()) {
+                for (int64_t pos = index * channels, i = 0; i < channels; ++i, ++pos) {
+                    new_order.emplace_back(pos);
+                }
+            }
+            assert(new_order.size() == static_cast<size_t>(output_channels));
+
+            ric.set_order(new_order);
+            ric_attr::set(conv->output(0), ric);
+            return true;
+        };
+
+        auto m = std::make_shared<pattern::Matcher>(pattern_root, matcher_name);
+        register_matcher(m, callback);
+    }
+};
+
+class ShapeOf : public ngraph::pass::MatcherPass {
+public:
+    ShapeOf() {
+        MATCHER_SCOPE(ShapeOf);
+        auto pattern_root = pattern::wrap_type<opset1::ShapeOf, opset8::ShapeOf>();
+
+        auto callback = [=](pattern::Matcher& m) {
+            // Skip propagation for ShapeOf path
```

Comment:
as discussed, it can be covered with test to verify that RIC is completely removed from graph

Reply:
test added.